### PR TITLE
Add basic set of base ubi9.4-minimal

### PR DESCRIPTION
**Pull request by ylebi**
Fix https://github.com/scylladb/scylla-operator-images/issues/13

### CODE REVIEW

**Review by tnozicka — COMMENTED** [review]
While I appreciate the initiative, let's discuss the overall plan first.
/hold

This will add significant build times so I'd like to make sure we use it.

And we need to figure our how we setup extra repositories, e.g. fedora or rocky for extra packages like `xfsprogs`
https://github.com/scylladb/scylla-operator-images/blob/196ac01/node-setup/Dockerfile#L4

**Comment by mykaul** [comment]
> Partially fix #13

Refs #13

File: base/ubi/9.4-minimal/Dockerfile
Comment (tnozicka) [inline base/ubi/9.4-minimal/Dockerfile:1]: given this builds `base-ubi-9.4-minimal` (rolling) this ref should be rolling as well - it also helps with periodics automatically picking the updates

File: base/ubi/9.4-minimal/Dockerfile
Comment (tnozicka) [inline base/ubi/9.4-minimal/Dockerfile:7]: this also used to explicitly install `curl` and `ca-certificates` - are you dropping them intentionally, or relying on it being already present?

File: golang/1.22/Dockerfile
Comment (tnozicka) [inline golang/1.22/Dockerfile:11]:
```suggestion
    microdnf install -y git make tar gzip gcc g++ && \
```

File: image-spec.csv
Comment (tnozicka) [inline image-spec.csv:3]: i don't think we need to build it anymore, given this switches the dependent images as well

File: base/ubi/9.4-minimal/Dockerfile
Comment (tnozicka) [reply base/ubi/9.4-minimal/Dockerfile:7]: if the later, stating the dependencies explicitly makes it more resilient to changes

File: node-setup/Dockerfile
Comment (tnozicka) [inline node-setup/Dockerfile:5]: I think we should do it in a base image so we don't have to repeat this in the dependent images. There could be more than one. It can possibly be disabled by default if you want to be more explicit about choosing it and enabling it when you really want it on demand with `--enablerepo`.

Comment (tnozicka) [inline node-setup/Dockerfile:5]: nit: EOF style might be more readable

**Comment by tnozicka** [comment]
also please clean / squash the commits

File: base/ubi/9.4-minimal/Dockerfile
Comment (ylebi) [reply base/ubi/9.4-minimal/Dockerfile:7]: curl and ca-certificates are already present.

File: base/ubi/9.4-minimal/Dockerfile
Comment (ylebi) [reply base/ubi/9.4-minimal/Dockerfile:1]: ok

File: node-setup/Dockerfile
Comment (ylebi) [reply node-setup/Dockerfile]: moved to a separate file

File: base/ubi/9.4-minimal/almalinux.repo
Comment (tnozicka) [inline base/ubi/9.4-minimal/almalinux.repo:1]:
```suggestion
[base-9]
```
when using `--enablerepo base-9` it might need a bit more specificity, like `alma-base-9`

**Review by tnozicka — APPROVED** [review]
thanks for the updates!

/approve
lgtm

@zimnx or @rzetelskik I am leaving the lgtm to one of you so this has a second set of eyes

File: node-setup/Dockerfile
Comment (zimnx) [inline node-setup/Dockerfile:4]: I recall mdadm had completely different behavior and command line parameters between versions. Just leaving a note in case e2e's around local disk setup would  failing after merging this.

**Comment by zimnx** [comment]
Oh we have two lgtm, so tagging it.
/lgtm

**Pull request merged**
